### PR TITLE
[SYCL][Doc] Update data flow pipes extension

### PR DESCRIPTION
**Pull request by mkinsner**
Update data flow pipes extension to be based on SYCL2020, and to make clear that pipe operations behave as if they are relaxed atomic operations.

Signed-off-by: Michael Kinsner <michael.kinsner@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
@DenisBakhvalov, `[SYCL :: ESIMD/matrix_transpose.cpp]` failure can't be related to this change. Could you take a look, please?

File: sycl/doc/extensions/DataFlowPipes/data_flow_pipes.asciidoc [hunk 1]
```diff
@@ -424,7 +424,9 @@ Pipes in the context of this extension step outside the OpenCL and SYCL memory m
 
 . There is no implicit synchronizes-with relationship between different pipes and/or with non-pipe memory in a named address space (e.g. global, local, private).  Specifically, there is no implicit global or local release of side effects through a pipe access, and observation of data or control information on one pipe does not imply any knowledge through happens-before relationship with a different pipe or with memory not associated with the pipe.
 
-. At a work-group barrier, there is an implicit acquire and release of side effects for any pipes operated on within the kernel, either before or after the barrier.  This occurs without an explicit pipe memory fence being applied as a barrier operand.
+. Pipe read and write operations behave as if they are C++ relaxed atomic load and store operations.  When paired with sycl::atomic_fences to establish a sychronizes-with relationship, pipe operations can provide guarantees on side effect visibility in memory, as defined by the C++ memory model.
```
Comment (Pennycook) [inline sycl/doc/extensions/DataFlowPipes/data_flow_pipes.asciidoc:427]: Nit: Should this be "SYCL relaxed atomic" and "SYCL memory model"?

**Review by Pennycook — DISMISSED** [review]
LGTM.  I have a question but consider it non-blocking.

File: sycl/doc/extensions/DataFlowPipes/data_flow_pipes.asciidoc (hunk 1 above)
Comment (mkinsner) [reply sycl/doc/extensions/DataFlowPipes/data_flow_pipes.asciidoc:427]: I agree.  Will update

**Review by Pennycook — APPROVED** [review]

**Comment by gmlueck** [comment]
If your intention is to bring this extension spec into conformance with SYCL 2020, I think more needs to be done:

* Define a feature-test macro
* Define the extension in the proper namespace
* Possibly update other naming to be in conformance with SYCL 2020 extension guidelines
* Update the specification format.  We are using the [device-if extension spec](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DeviceIf/device_if.asciidoc) as a model template.

**Comment by gmlueck** [comment]
One more comment: I think we should **not** commit the HTML rendered version of these specs into GitHub.  I've made this comment for other specs too.  This just opens the possibility of the HTML and Asciidoc versions getting out of sync, contributing to confusion.  Instead, just rely on the GitHub preview feature to view the Asciidoc version.  I find this actually works quite well.

**Comment by mkinsner** [comment]
> If your intention is to bring this extension spec into conformance with SYCL 2020, I think more needs to be done:
> 
> * Define a feature-test macro
> * Define the extension in the proper namespace
> * Possibly update other naming to be in conformance with SYCL 2020 extension guidelines
> * Update the specification format.  We are using the [device-if extension spec](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/DeviceIf/device_if.asciidoc) as a model template.

I'd like to make these changes independently when we update the other specs, unless there is a reason to do it now.  The change in this iteration is a clarification and doesn't require any implementation change - I'd prefer that we batch the feature test macro updates, for example.

**Comment by mkinsner** [comment]
> One more comment: I think we should **not** commit the HTML rendered version of these specs into GitHub. I've made this comment for other specs too. This just opens the possibility of the HTML and Asciidoc versions getting out of sync, contributing to confusion. Instead, just rely on the GitHub preview feature to view the Asciidoc version. I find this actually works quite well.

Agreed.  Removing.

**Comment by gmlueck** [comment]
> I'd like to make these changes independently when we update the other specs, unless there is a reason to do it now. The change in this iteration is a clarification and doesn't require any implementation change - I'd prefer that we batch the feature test macro updates, for example.

OK.

**Pull request merged**